src/memory/AmdFlash.cc

```diff
@@ -780,6 +792,55 @@ void AmdFlash::EraseOperation::reset()
 	buffer.clear();
 }
 
+bool AmdFlash::checkCommandSuspend(EmuTime::param time)
+{
+	if (state == State::ERASE_SECTOR) {
+		if (chip.erase.suspend) {
+			if (cmd[0].value == 0xB0) {
+				if (!status.eraseTimer) {
+					syncOperation.removeSyncPoint();
+					execSuspend(time);
+				} else {
+					if (!syncSuspend.pendingSyncPoint()) {
+						syncOperation.removeSyncPoint();
+						syncSuspend.setSyncPoint(time + chip.erase.suspendDuration);
+					}
+				}
+			}
+		}
+	} else {
+		UNREACHABLE;
+	}
+	return false;
+}
+
+void AmdFlash::execSuspend(EmuTime::param /*time*/)
+{
+	if (state == State::ERASE_SECTOR) {
+		statusRegister.ready = true;
+		statusRegister.eraseSuspend = true;
+		status.dataPolling = true;
+		status.eraseTimer = true;
+		setState(State::IDLE);
+	} else {
+		UNREACHABLE;
```

Comment:
This is fine. But isn't it simpler to use `assert(state == State::ERASE_SECTOR);`?
Ans similar for some other places where you use `UNREACHABLE`.